packages/search-in-workspace/src/browser/search-in-workspace-widget.tsx

```diff
@@ -316,19 +316,30 @@ export class SearchInWorkspaceWidget extends BaseWidget implements StatefulWidge
 
     protected renderReplaceFieldToggle(): React.ReactNode {
         const toggle = <span className={`fa fa-caret-${this.showReplaceField ? 'down' : 'right'}`}></span>;
+
+        const replaceToggle = () => {
+            const elArr = document.getElementsByClassName('replace-toggle');
+            if (elArr && elArr.length > 0) {
+                (elArr[0] as HTMLElement).focus();
+            }
+            this.showReplaceField = !this.showReplaceField;
+            this.resultTreeWidget.showReplaceButtons = this.showReplaceField;
+            this.update();
+        };
+
         return <div
             title='Toggle Replace'
             className='replace-toggle'
             tabIndex={0}
-            onClick={e => {
-                const elArr = document.getElementsByClassName('replace-toggle');
-                if (elArr && elArr.length > 0) {
-                    (elArr[0] as HTMLElement).focus();
```

Comment:
Let's clean this up:
```ts
export class SearchInWorkspaceWidget extends BaseWidget implements StatefulWidget {
  // Do not touch it, use react to modify the DOM
  protected replaceToggleRef: HTMLElement | undefined;
  protected setReplaceToggleRef = (element: HTMLElement | null) => {
    if (element) this.replaceToggleRef = element;
  };

  // enhance the replace toggle
  protected renderReplaceFieldToggle(): React.ReactNode {
    // other stuff
    return <div
      title='Toggle Replace'
      className='replace-toggle'
      ref={this.setReplaceToggleRef} // <-- this is the new thing
      tabIndex={0}
      ...
  }
}
```

And then, you can get rid of:
```ts
const elArr = document.getElementsByClassName('replace-toggle');
if (elArr && elArr.length > 0) {
  (elArr[0] as HTMLElement).focus();
}
```

And replace it with:
```ts
if (this.replaceToggleRef) {
  replaceToggleRef.focus();
}
```